### PR TITLE
Fedora packaging

### PR DESCRIPTION
**Pull request by M0ses**
This is first raw version - commits should be rebased or stashed together

### CODE REVIEW

**Comment by coolo** [comment]
how about you have the suse packages succeed before you move forward? :)

File: dist/obs-bundled-gems.spec [hunk 1]
```diff
@@ -105,18 +127,20 @@ bundle config build.ffi --enable-system-libffi
 bundle config build.nokogiri --use-system-libraries
 bundle config build.sassc --disable-march-tune-native
 
-%install
```
Comment (Conan-Kudo) [inline dist/obs-bundled-gems.spec]: You accidentally deleted the `%install` section

File: dist/setup-appliance.sh [hunk 2]
```diff
@@ -580,6 +591,42 @@ EOF
 
 }
 
+function prepare_os_settings {
+  . /etc/os-release
+  for d in $ID_LIKE;do
```
Comment (Conan-Kudo) [inline dist/setup-appliance.sh]: `ID_LIKE` doesn't exist in Fedora's `os-release` file. You'll need to concatenate `$ID` and `$ID_LIKE` into a single array first.

File: dist/obs-bundled-gems.spec (hunk 1 above)
Comment (M0ses) [reply dist/obs-bundled-gems.spec]: No, not accidentally.

Otherwise the built gems failed.

**Comment by M0ses** [comment]
> how about you have the suse packages succeed before you move forward? :)

Should be fixed now - but O:S:U/obs-server needs an additional extract line - this is why the test failing here, but works in branch

https://build.opensuse.org/package/show/home:M0ses:branches:OBS:Server:Unstable/obs-server

For the record:

```
openSUSE_Factory     x86_64     obs-server                     succeeded*
openSUSE_Factory     x86_64     obs-server:obs-bundled-gems    succeeded*
openSUSE_15.1        x86_64     obs-server                     succeeded*
openSUSE_15.1        x86_64     obs-server:obs-bundled-gems    succeeded*
openSUSE_15.0        x86_64     obs-server                     succeeded*
openSUSE_15.0        x86_64     obs-server:obs-bundled-gems    succeeded*
SLE_15_SP1           x86_64     obs-server                     succeeded*
SLE_15_SP1           x86_64     obs-server:obs-bundled-gems    succeeded*
Fedora_Rawhide       x86_64     obs-server                     unresolvable*
Fedora_Rawhide       x86_64     obs-server:obs-bundled-gems    failed*
Fedora_31            x86_64     obs-server                     succeeded*
Fedora_31            x86_64     obs-server:obs-bundled-gems    succeeded*
CentOS_8             x86_64     obs-server                     unresolvable*
CentOS_8             x86_64     obs-server:obs-bundled-gems    unresolvable*
CentOS_7             x86_64     obs-server                     unresolvable*
CentOS_7             x86_64     obs-server:obs-bundled-gems    failed*
```

File: dist/setup-appliance.sh (hunk 2 above)
Comment (M0ses) [reply dist/setup-appliance.sh]: In my test installation it is in there.

Taken from my test installation: 

```
[kanku@obs-server ~]$ cat /etc/os-release 
NAME=Generic
VERSION="31 (Generic)"
ID=generic
ID_LIKE=fedora
...
[kanku@obs-server ~]$ rpm -qf /etc/os-release
generic-release-common-31-2.noarch
[kanku@obs-server ~]$ rpm -qi generic-release-common-31-2.noarch
Name        : generic-release-common
...
Build Host  : buildhw-10.phx2.fedoraproject.org
Packager    : Fedora Project
Vendor      : Fedora Project
...
[kanku@obs-server ~]$

```

Comment (Conan-Kudo) [reply dist/setup-appliance.sh]: `generic-release` has it, but `fedora-release` will not.

Comment (Conan-Kudo) [reply dist/setup-appliance.sh]: For example, from my computer:

```
[neal@yu-ohki ~]$ cat /etc/os-release
NAME=Fedora
VERSION="31 (KDE Plasma)"
ID=fedora
VERSION_ID=31
VERSION_CODENAME=""
PLATFORM_ID="platform:f31"
PRETTY_NAME="Fedora 31 (KDE Plasma)"
ANSI_COLOR="0;34"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:31"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f31/system-administrators-guide/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=31
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=31
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="KDE Plasma"
VARIANT_ID=kde
```

**Pull request merged**